inference-engine/src/mkldnn_plugin/nodes/detectionoutput.cpp

```diff
@@ -121,60 +123,92 @@ class DetectionOutputImpl: public ExtLayerBase {
                        ResponseDesc *resp) noexcept override {
         float *dst_data = outputs[0]->buffer();
 
-        const float *loc_data    = inputs[idx_location]->buffer();
-        const float *conf_data   = inputs[idx_confidence]->buffer();
-        const float *prior_data  = inputs[idx_priors]->buffer();
+        const float *loc_data    = inputs[idx_location]->buffer().as<const float *>();
+        const float *conf_data   = inputs[idx_confidence]->buffer().as<const float *>();
+        const float *prior_data  = inputs[idx_priors]->buffer().as<const float *>();
+        const float *arm_conf_data = inputs.size() > 3 ? inputs[idx_arm_confidence]->buffer().as<const float *>() : nullptr;
+        const float *arm_loc_data = inputs.size() > 4 ? inputs[idx_arm_location]->buffer().as<const float *>() : nullptr;
 
         const int N = inputs[idx_confidence]->getTensorDesc().getDims()[0];
 
-        float *decoded_bboxes_data = _decoded_bboxes->buffer();
-        float *reordered_conf_data = _reordered_conf->buffer();
-        float *bbox_sizes_data     = _bbox_sizes->buffer();
-        int *detections_data       = _detections_count->buffer();
-        int *buffer_data           = _buffer->buffer();
-        int *indices_data          = _indices->buffer();
-        int *num_priors_actual     = _num_priors_actual->buffer();
+        float *decoded_bboxes_data = _decoded_bboxes->buffer().as<float *>();
+        float *reordered_conf_data = _reordered_conf->buffer().as<float *>();
+        float *bbox_sizes_data     = _bbox_sizes->buffer().as<float *>();
+        int *detections_data       = _detections_count->buffer().as<int *>();
+        int *buffer_data           = _buffer->buffer().as<int *>();
+        int *indices_data          = _indices->buffer().as<int *>();
+        int *num_priors_actual     = _num_priors_actual->buffer().as<int *>();
 
         for (int n = 0; n < N; ++n) {
             const float *ppriors = prior_data;
             const float *prior_variances = prior_data + _num_priors*_prior_size;
             if (_priors_batches) {
                 ppriors += _variance_encoded_in_target ? n*_num_priors*_prior_size : 2*n*_num_priors*_prior_size;
-                prior_variances += _variance_encoded_in_target ? 0 : n*_num_priors*_prior_size;
+                prior_variances += _variance_encoded_in_target ? 0 : 2*n*_num_priors*_prior_size;
             }
 
             if (_share_location) {
                 const float *ploc = loc_data + n*4*_num_priors;
                 float *pboxes = decoded_bboxes_data + n*4*_num_priors;
                 float *psizes = bbox_sizes_data + n*_num_priors;
-                decodeBBoxes(ppriors, ploc, prior_variances, pboxes, psizes, num_priors_actual, n);
+
+                if (inputs.size() > 4) {
```

Comment:
Does it make sense to change these kind of checks on smt meaningful like "with_additional_box_predictions"?

Reply:
changed